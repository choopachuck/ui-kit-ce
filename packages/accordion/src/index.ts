export * from './Accordion'
export * from './AccordionItem'
export type {
  AccordionIconProps,
  AccordionComponentsConfig,
} from './components'
