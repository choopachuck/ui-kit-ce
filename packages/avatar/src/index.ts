export * from './Avatar'
export * from './AvatarGroup'
export * from './types'
export type {
  AvatarGroupExtraProps,
  AvatarGroupComponentsConfig,
} from './components'
export * from './classes'
export * from './constants'
