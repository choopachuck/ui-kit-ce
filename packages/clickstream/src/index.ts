export * from './ClickStreamProvider'
export type {
  ClickStreamBaseData,
  ClickStreamContextProps,
  ClickStreamElementProperty,
  ClickStreamElementType,
  ClickStreamEventBatch,
  ClickStreamEventBatchItem,
  ClickStreamEventData,
  ClickStreamEventMeta,
  ClickStreamProviderProps,
} from './types'
