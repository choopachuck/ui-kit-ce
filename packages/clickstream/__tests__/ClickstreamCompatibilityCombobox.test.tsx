/*--------------------------------------------------------------------------------------------------
 * --- Тестирование совместимости компонентов ComboBox и Autocomplete с провайдером ClickStream ----
 * -------------------------------------------------------------------------------------------------
 */

import * as React from 'react'
import { render, waitFor } from '@testing-library/react'
import { ClickStreamComponentsTest } from '../examples/ClickStreamComponentsTest'
import * as ClickStreamUtils from '../src/utils'
import userEvent from '@testing-library/user-event'
import { mockGlobals } from './helpers'

jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01'))
jest
  .spyOn(ClickStreamUtils, 'getISOTime')
  .mockImplementation(() => '2020-01-01T00:00:00.078-00:00')

const pause = async (ms: number) => {
  jest.advanceTimersByTime(ms)
  await Promise.resolve()
}

beforeAll(() => {
  mockGlobals()
})

it('clickstream lib components compatibility: combobox::single', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 3

  const { getByTitle, findAllByRole } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const combobox = getByTitle('combobox')

  await waitFor(() => userEvent.click(combobox), { timeout: 1 })
  const foundOptions = await findAllByRole('option')

  await waitFor(() => userEvent.click(foundOptions[1]), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: combobox::multiple', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 15
  const searchValue = 'Треть'

  const { getByTitle, findAllByRole } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const combobox = getByTitle('combobox-multiple')
  const searchInput = getByTitle('combobox-multiple-search-input')

  await waitFor(() => userEvent.click(combobox), { timeout: 1 })
  let foundOptions = await findAllByRole('option')

  await waitFor(() => userEvent.click(foundOptions[1]), { timeout: 1 })
  await waitFor(() => userEvent.click(foundOptions[2]), { timeout: 1 })
  await waitFor(() => userEvent.click(foundOptions[2]), { timeout: 1 })
  await waitFor(() => userEvent.click(searchInput), { timeout: 1 })
  userEvent.keyboard(searchValue)

  foundOptions = await findAllByRole('option')

  await waitFor(() => userEvent.click(foundOptions[0]), { timeout: 1 })

  await waitFor(() => userEvent.click(combobox), { timeout: 1 })

  for (let i = 0; i <= searchValue.length; i++) {
    userEvent.keyboard('[Backspace]')
  }

  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() => userEvent.click(combobox), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: combobox::creatable', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 4
  const newOption = 'Новая опция'

  const { getByTitle, findAllByRole } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const comboboxInput = getByTitle('combobox-creatable-input')

  await waitFor(() => userEvent.click(comboboxInput), { timeout: 1 })
  userEvent.keyboard(newOption)
  const foundOptions = await findAllByRole('option')
  await waitFor(() => userEvent.click(foundOptions[0]), { timeout: 1 })

  await waitFor(() => userEvent.tab(), { timeout: 1 })

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})

it('clickstream lib components compatibility: autocomplete::keyboard', async () => {
  const handleBatch = jest.fn()
  const handleSendEvent = jest.fn()
  const expectedBatchCalls = 6

  const { getByTitle } = render(
    <ClickStreamComponentsTest
      batchSize={expectedBatchCalls}
      onSendEvent={handleSendEvent}
      onBatch={handleBatch}
    />
  )

  const searchInput = getByTitle(
    'autocomplete-search-input'
  ) as HTMLInputElement

  await waitFor(() => userEvent.click(searchInput), { timeout: 1 })

  userEvent.keyboard('о')

  await pause(100)

  userEvent.keyboard('[ArrowDown]')

  await pause(200)

  userEvent.keyboard('[ArrowDown]')

  await pause(100)

  userEvent.keyboard('[ArrowDown]')

  await pause(100)

  userEvent.keyboard('[ArrowDown]')

  await pause(100)

  userEvent.keyboard('[ArrowUp]')

  await pause(100)

  await waitFor(() =>
    expect(handleBatch).toHaveBeenCalledTimes(expectedBatchCalls)
  )
  await waitFor(() => expect(handleSendEvent).toMatchSnapshot())
})
