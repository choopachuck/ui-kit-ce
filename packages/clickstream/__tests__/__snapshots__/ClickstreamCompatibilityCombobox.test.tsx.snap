// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clickstream lib components compatibility: autocomplete::keyboard 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-88",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-90",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "autocomplete-search-input",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "autocomplete",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Пустое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Пустое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "autocomplete",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Второе",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Второе",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "autocomplete",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Четвертое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Четвертое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "autocomplete",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Пятое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Пятое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "autocomplete",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Четвертое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Четвертое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: combobox::creatable 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-59",
              },
              Object {
                "key": "aria-expanded",
                "value": "true",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-61",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "combobox-creatable-input",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box-creatable",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "\\"Новая опция\\"",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "\\"Новая опция\\"",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "false",
              },
              Object {
                "key": "id",
                "value": "v-uik-63-option-Новая опция",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "\\"Новая опция\\"",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-59",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-61",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "combobox-creatable-input",
              },
              Object {
                "key": "value",
                "value": "Новая опция",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Новая опция",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: combobox::multiple 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "rootControl-0-2-143 medium-0-2-148 focused-0-2-144 isSearchable-0-2-150",
              },
              Object {
                "key": "aria-expanded",
                "value": "true",
              },
              Object {
                "key": "aria-labelledby",
                "value": "v-uik-34",
              },
              Object {
                "key": "role",
                "value": "combobox",
              },
              Object {
                "key": "tabindex",
                "value": "-1",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple",
              },
            ],
            "elementEvent": "click",
            "elementTag": "div",
            "elementType": "div",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Поиск",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "true",
              },
              Object {
                "key": "id",
                "value": "v-uik-35-option-1",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "optionActive-0-2-133 listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое, Второе",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое, Второе",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "true",
              },
              Object {
                "key": "id",
                "value": "v-uik-35-option-2",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "optionActive-0-2-133 listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Второе",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "false",
              },
              Object {
                "key": "id",
                "value": "v-uik-35-option-2",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "optionActive-0-2-133 listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Второе",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-31",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-33",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple-search-input",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "click",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое, Третье",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое, Третье",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "true",
              },
              Object {
                "key": "id",
                "value": "v-uik-35-option-3",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "optionActive-0-2-133 listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Третье",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-31",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-33",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple-search-input",
              },
              Object {
                "key": "value",
                "value": "Треть",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Треть",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "rootControl-0-2-143 medium-0-2-148 focused-0-2-144 isSearchable-0-2-150",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "aria-labelledby",
                "value": "v-uik-34",
              },
              Object {
                "key": "role",
                "value": "combobox",
              },
              Object {
                "key": "tabindex",
                "value": "-1",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple",
              },
            ],
            "elementEvent": "click",
            "elementTag": "div",
            "elementType": "div",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "ПервоеТретье",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "input-0-2-163",
              },
              Object {
                "key": "aria-autocomplete",
                "value": "list",
              },
              Object {
                "key": "aria-controls",
                "value": "v-uik-31",
              },
              Object {
                "key": "aria-expanded",
                "value": "false",
              },
              Object {
                "key": "autocapitalize",
                "value": "none",
              },
              Object {
                "key": "autocomplete",
                "value": "off",
              },
              Object {
                "key": "autocorrect",
                "value": "off",
              },
              Object {
                "key": "id",
                "value": "v-uik-33",
              },
              Object {
                "key": "spellcheck",
                "value": "false",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "type",
                "value": "text",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple-search-input",
              },
              Object {
                "key": "value",
                "value": "",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "text",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "rootControl-0-2-143 medium-0-2-148 focused-0-2-144 isSearchable-0-2-150",
              },
              Object {
                "key": "aria-expanded",
                "value": "true",
              },
              Object {
                "key": "aria-labelledby",
                "value": "v-uik-34",
              },
              Object {
                "key": "role",
                "value": "combobox",
              },
              Object {
                "key": "tabindex",
                "value": "-1",
              },
              Object {
                "key": "title",
                "value": "combobox-multiple",
              },
              Object {
                "key": "aria-activedescendant",
                "value": "v-uik-35-option-Первое",
              },
            ],
            "elementEvent": "click",
            "elementTag": "div",
            "elementType": "div",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`clickstream lib components compatibility: combobox::single 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "class",
                "value": "rootControl-0-2-143 medium-0-2-148 focused-0-2-144",
              },
              Object {
                "key": "aria-expanded",
                "value": "true",
              },
              Object {
                "key": "aria-labelledby",
                "value": "v-uik-6",
              },
              Object {
                "key": "role",
                "value": "combobox",
              },
              Object {
                "key": "tabindex",
                "value": "0",
              },
              Object {
                "key": "title",
                "value": "combobox",
              },
            ],
            "elementEvent": "click",
            "elementTag": "div",
            "elementType": "div",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Выберите опцию",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "role",
                "value": "option",
              },
              Object {
                "key": "aria-selected",
                "value": "false",
              },
              Object {
                "key": "id",
                "value": "v-uik-7-option-1",
              },
              Object {
                "key": "aria-disabled",
                "value": "false",
              },
              Object {
                "key": "class",
                "value": "listItem-0-2-212 option-0-2-127 medium-0-2-218 optionMedium-0-2-129",
              },
            ],
            "elementEvent": "click",
            "elementTag": "li",
            "elementType": "li",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": MouseEvent {
            "isTrusted": false,
          },
        },
        Object {
          "data": Object {
            "batteryLevel": "0",
            "connectionType": "UNKNOWN",
            "elementAttributes": Array [
              Object {
                "key": "data-v-uik-input-type",
                "value": "combo-box",
              },
              Object {
                "key": "type",
                "value": "hidden",
              },
              Object {
                "key": "value",
                "value": "Первое",
              },
            ],
            "elementEvent": "change",
            "elementTag": "input",
            "elementType": "hidden",
            "pageName": "",
            "timeStamp": "2020-01-01T00:00:00.078-00:00",
            "url": "http://localhost/",
            "value": "Первое",
          },
          "event": Event {
            "isTrusted": false,
          },
        },
      ],
      Object {
        "applicationLanguage": "ru",
        "browser": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/16.7.0",
        "platform": "WEB",
        "screenSize": "0x0",
        "systemLanguage": "ru",
        "timeStamp": "2020-01-01T00:00:00.078-00:00",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
